Tests/EmbeddedScripts.CSharp.MonoEvaluator.Tests/MonoCSharpRunnerTests.cs

```diff
@@ -0,0 +1,226 @@
+        {
+            var s = "abc";
+            var t = new HelperObject();
+            var code = "t.x += s.Length;";
+
+            var runner = new MonoCSharpRunner()
+                .Register(s, "s");
+
+            await Assert.ThrowsAsync<ScriptSyntaxErrorException>(() => runner.RunAsync(code));
+
+            runner.Register(t, "t");
+
+            await runner.RunAsync(code);
+        }
+
+        [Fact]
+        public async Task RunWithTwoGlobalVariables_Succeed()
+        {
+            var runner = new MonoCSharpRunner()
+                .Register(1, "a")
+                .Register(2, "b");
+
+            await runner.RunAsync("var c = a + b;");
+        }
+
+        [Fact]
+        public async Task RunWithGlobalFunc_Succeed()
+        {
+            int x = 0;
+            var code = "t();";
+
+            var runner = new MonoCSharpRunner()
+                .Register<Action>(() => { x++; }, "t");
+
+            await runner.RunAsync(code);
+
+            Assert.Equal(1, x);
+        }
+
+        [Fact]
+        public async Task RunInvalidCode_ThrowsException()
+        {
+            var code = "a = 1;";
+
+            var runner = new MonoCSharpRunner();
+
+            await Assert.ThrowsAsync<ScriptSyntaxErrorException>(() => runner.RunAsync(code));
+        }
+
+        [Fact]
+        public async void RunCodeWithRuntimeError_ThrowsAnException()
+        {
+            var code = "int a = 1; int b = 2 / (a - a);";
+
+            var runner = new MonoCSharpRunner();
+
+            await Assert.ThrowsAsync<DivideByZeroException>(() => runner.RunAsync(code));
+        }
+
+        [Fact]
+        public async Task ExposedFuncThrowingException_RunnerThrowsSameException()
+        {
+            var exceptionMessage = "hello from exception";
+            var runner = new MonoCSharpRunner()
+                .Register<Action>(() => throw new ArgumentException(exceptionMessage), "f");
+
+            var exception = await Assert.ThrowsAsync<ArgumentException>(() =>
+                runner.RunAsync("f()"));
+
+            Assert.Equal(exceptionMessage, exception.Message);
+        }
+
+        [Fact]
+        public async void CodeThrowsAnException_SameExceptionIsThrowingFromRunner()
+        {
+            var exceptionMessage = "Exception from user code";
+            var code = $"throw new System.ArgumentException(\"{exceptionMessage}\");";
+
+            var exception = await Assert.ThrowsAsync<ArgumentException>(() =>
+                new MonoCSharpRunner().RunAsync(code));
+
+            Assert.Equal(exceptionMessage, exception.Message);
+        }
+
+        [Fact]
+        public async Task ExposeFunc_Succeed()
+        {
+            await new MonoCSharpRunner()
+                .Register<Func<int, int, int>>((a, b) => a + b, "Add")
+                .RunAsync("var c = Add(1, 2);");
+        }
+
+        [Fact]
+        public async Task RunAsyncWithContinuation_EachRunSharesGlobals_Success()
```

Comment:
nit: if you use specific naming pattern for test names (like 3 parts here), then it would be better to follow that pattern for all test namings